packages/aws-cdk-lib/aws-dynamodb/lib/table-v2-base.ts

```diff
@@ -431,11 +431,11 @@ export abstract class TableBaseV2 extends Resource implements ITableV2, IResourc
     if (options.tableActions) {
       const resourceArns = [this.tableArn];
       this.hasIndex && resourceArns.push(`${this.tableArn}/index/*`);
-      return Grant.addToPrincipalOrResource({
+      return Grant.addToPrincipal({
```

Comment:
Why is this the right fix? AFAIK, `addToPrincipalOrResource` will first call `addToPrincipal` first anyway. My guess of the root cause is that users can call `addToResourcePolicy` method on tableV2 construct, but `resourcePolicy` is not created as a lazy variable, thus it will only have the resource policy when construct is first defined.